### PR TITLE
feat: [CI-6657]: allow expression for codebase reponame

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

allow expression for codebase reponame

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by shaurya-harness** [comment]
retrigger jest

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx [hunk 1]
```diff
@@ -631,6 +631,12 @@ function CICodebaseInputSetFormInternal({
       : [MultiTypeInputType.FIXED]
   }, [viewTypeMetadata?.isTemplateBuilder])
 
+  const AllowableTypesForCodebaseRepoName = useMemo((): AllowedTypesWithRunTime[] => {
+    return viewTypeMetadata?.isTemplateBuilder || viewTypeMetadata?.isTemplateDetailDrawer
+      ? [MultiTypeInputType.RUNTIME, MultiTypeInputType.FIXED, MultiTypeInputType.EXPRESSION]
+      : [MultiTypeInputType.FIXED, MultiTypeInputType.EXPRESSION]
+  }, [viewTypeMetadata?.isTemplateBuilder])
```
Comment (d1wilko) [inline src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:638]: should this include `viewTypeMetadata?.isTemplateDetailDrawer`?

**Review by d1wilko — CHANGES_REQUESTED** [review]
one small question - otherwise looks good

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx (hunk 1 above)
Comment (shaurya-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:638]: yes we have been using the same for all codebase properties , check AllowableTypesForCodebaseProperties at line 628

Comment (d1wilko) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:638]: Sorry I mean should the array at the end of the useMemo not also include `viewTypeMetadata?.isTemplateDetailDrawer`

**Review by d1wilko — CHANGES_REQUESTED** [review]

**Review by d1wilko — APPROVED** [review]
LGTM :)

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx (hunk 1 above)
Comment (shaurya-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx]: right, updated

**Comment by shaurya-harness** [comment]
retrigger jest

**Pull request merged**